packages/design-tokens-schema/src/walker.ts

```diff
@@ -7,13 +7,13 @@ import { ColorToken, ColorTokenValidationSchema } from './color-token';
  */
 export const walkObject = <T = unknown>(
   data: unknown,
-  predicate: (data: unknown) => data is T,
-  callback: (data: T, path: string[]) => void,
+  predicate: (data: unknown, path: string[]) => data is T,
+  callback?: (data: T, path: string[]) => void,
 ): void => {
   function traverse(currentData: unknown, path: string[]): void {
     // Check if current data matches
-    if (predicate(currentData)) {
-      callback(currentData, path);
+    if (predicate(currentData, path)) {
+      callback?.(currentData, path);
```

Comment:
if the callback is optional, what does the function do? Its return type is `void`

Reply:
For the ref validation the callback is optional because the predicate throws Errors that we process in `superRefine()`. I had a `() => {}` here at first, but figured this was cleaner.